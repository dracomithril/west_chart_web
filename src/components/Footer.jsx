/**
 * Created by Gryzli on 12.02.2017.
 */
import React from 'react';

const Footer = () => (
  <footer className="footer">
    <span>
      {'site created by '}
      <a href="https://github.com/dracomithril" target="_newtab">
        dracomithril{' '}
      </a>
      repo on
      <a href="https://github.com/dracomithril/WestChart_ReactApp" target="_newtab">
        <i className="fa fa-github" aria-hidden="true" />
        <br />
      </a>
      {` v${process.env.REACT_APP_VERSION}`}
      <i className="fa fa-copyright" />
      {' Copyright 2017'}
    </span>
    <br />
    <span>
      {'Any questions? '}
      <a href="mailto:dracomithril@gmail.com?subject=[WCSChartAdmin]">
        <i className="fa fa-envelope" />
        {' contact me'}
      </a>
    </span>
    <br />
    <a href="/policy">
      <i className="fa fa-facebook-square" />
      {' Policy'}
    </a>
  </footer>
);

export default Footer;
