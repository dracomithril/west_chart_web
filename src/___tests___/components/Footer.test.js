/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import Footer from './../../components/Footer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureMockStore([]);
describe('<Footer/>', () => {
  it('renders without crashing', () => {
    process.env.npm_package_version = "0.6.3";
    const store = mockStore({});
    const wrapper = shallow(
      <Footer/>, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });
});
