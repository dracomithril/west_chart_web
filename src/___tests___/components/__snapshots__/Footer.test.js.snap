// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> renders without crashing 1`] = `
<footer
  className="footer"
>
  <span>
    site created by 
    <a
      href="https://github.com/dracomithril"
      target="_newtab"
    >
      dracomithril
       
    </a>
    repo on
    <a
      href="https://github.com/dracomithril/WestChart_ReactApp"
      target="_newtab"
    >
      <i
        aria-hidden="true"
        className="fa fa-github"
      />
      <br />
    </a>
     v0.6.3
    <i
      className="fa fa-copyright"
    />
     Copyright 2017
  </span>
  <br />
  <span>
    Any questions? 
    <a
      href="mailto:dracomithril@gmail.com?subject=[WCSChartAdmin]"
    >
      <i
        className="fa fa-envelope"
      />
       contact me
    </a>
  </span>
  <br />
  <a
    href="/api/fb/policy"
  >
    <i
      className="fa fa-facebook-square"
    />
     Policy
  </a>
</footer>
`;
